php-fpm/Dockerfile

```diff
@@ -1,57 +1,155 @@
+ENV USER_ID         501
+ENV GROUP_ID        20
+ENV OSTYPE          darwin220
 
-RUN dnf install -y which pv sudo bind-utils python3-pip bash-completion rsync socat oathtool \
-        wget ack make gcc gcc-c++ automake autoconf crontabs mariadb nano vim coreutils-common \
-    && dnf clean all \
-    && rm -rf /var/cache/dnf
+# PHP Extension Installer
+ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
 
-RUN dnf install -y python2 \
-    && ln -s /bin/python2 /bin/python \
-    && dnf clean all \
-    && rm -rf /var/cache/dnf
+# Copy mhsendmail to PHP
+COPY --from=mhs-builder /go/bin/mhsendmail /usr/local/bin/mhsendmail
 
-ENV NODE_VERSION    10
-RUN npm install -g n \
-    && n install ${NODE_VERSION} \
-    && rm -rf /usr/local/n/versions/node
+# Install PHP Extensions required by Magento OS, Adobe Commerce, and the UCT (pcntl)
+RUN chmod +x /usr/local/bin/install-php-extensions && install-php-extensions \
+    apcu \
+    amqp \
+    bcmath \
+    calendar \
+    exif \
+    gd \
+    intl \
+    imagick \
+    imap \
+    mcrypt \
+    mysqli \
+    pcntl \
+    pdo_mysql \
+    redis \
+    soap \
+    sockets \
+    sodium \
+    xsl \
+    zip
+
+# Install helpful utilities
+RUN set -eux; \
+    apt-get update -y; \
+    apt-get install -y --no-install-recommends \
+    jq \
+    bash \
+    patch \
+    pv \
+    procps \
+    nano \
+    vim \
+    mariadb-client \
+    sudo \
+    gettext \
+    ca-certificates \
+    git \
+    rsync \
+    bash-completion \
+    nmap  \
+    less \
+    zsh \
+    fish \
+    pwgen \
+    openssh-client \
+    unzip \
+    rsync \
+    socat \
+    oathtool \
+    wget \
+    ack  \
+    automake  \
+    autoconf \
+    python \
+    cron \
+    htop \
+    chromium \
+    imagemagick \
+    graphicsmagick
 
 COPY docker-entrypoint /usr/local/bin/
 COPY etc/profile.d/*.sh /etc/profile.d/
-COPY etc/*.ini /etc/
-COPY etc/php.d/*.ini /etc/php.d/
-COPY etc/php.d/05-additions.ini.template /etc/php.d/
+COPY etc/php.d/*.ini /usr/local/etc/php/conf.d
+COPY etc/php.d/05-additions.ini.template /usr/local/etc/php/conf.d
+
+RUN set -eux \
+    && cp ${PHP_INI_DIR}/php.ini-development ${PHP_INI_DIR}/php.ini \
+    && perl -pi -e 's/memory_limit(.*)/memory_limit=-1/g' ${PHP_INI_DIR}/php.ini
+
+# Install Composer
+COPY --from=composer:1 /usr/bin/composer /usr/bin/composer1
+COPY --from=composer:2.2 /usr/bin/composer /usr/bin/composer2lts
+COPY --from=composer:2 /usr/bin/composer /usr/bin/composer2
 
-# Install mhsendmail to support routing email through mailhog
-RUN mkdir -p /tmp/mhsendmail \
-    && cd /tmp/mhsendmail \
-    && curl -vLOs https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64 \
-    && chmod +x mhsendmail_linux_amd64 \
-    && mv mhsendmail_linux_amd64 /usr/local/bin/mhsendmail
+# If PHP < 7.2
+RUN if [ "$(printf "7.2\n${PHP_VERSION}" | sort -g | head -n1)" != "7.2" ]; \
```

Comment:
I would not do this. Since the COMPOSER_VERSION in the environment will determine which specific `composer*` phar to use, forcing LTS composer for _all_ PHP images above 7.2 removes some of that flexibility.

Reply:
@bap14 actually this sets the composer to lts version for php 7.0 and 7.1 not above, that is determined by the environment. when the container is started and the entrypoint is run see:
https://github.com/rbouma/warden-images/blob/6571c288cbe0b223b9f8df21373f80ab9d930e09/php-fpm/context/docker-entrypoint#L51

This is just a default setup and the env variable will determine the actual composer value.